### PR TITLE
#3280 pythagoras fix

### LINKED ISSUE #3280: Pythagoras block doesn't handle negative y2
With u1IsHypotenuse = true, Math.Blocks.Pythagoras doesn't properly handle the case where y2 goes negative. The code should use an if-expression (rather than an if-equation) wrapped in a noEvent to prevent evaluating the square root with a negative value. As it is, an event is created when y2 goes negative, but that doesn't prevent the problem.

### PR DESCRIPTION
**Pull request by JoeRiel**
My fix for #3280 (issue with Pythagoras block). Resolves #3280.

### CODE REVIEW

File: Modelica/Blocks/Math.mo [hunk 1]
```diff
@@ -3505,13 +3505,8 @@ so one can plot the result directly as frequency lines.
...
       y2 = u1^2-u2^2;
-      if y2>=0 then
-        y=sqrt(y2);
-        valid = true;
-      else
-        y=0;
-        valid = false;
-      end if;
+      valid = (y2 >= 0);  
+      y = noEvent(if y2 >= 0 then sqrt(y2) else 0);
```
Comment (beutlich) [inline Modelica/Blocks/Math.mo]: What about?

```
valid = noEvent(y2 >= 0);
y = if valid then sqrt(y2) else 0;
```

Comment (JoeRiel) [reply Modelica/Blocks/Math.mo]: That won't work, I believe, because valid is a discrete variable, so its equation must occur in an event.

Comment (JoeRiel) [reply Modelica/Blocks/Math.mo]: A nice variation would be
  `valid = y2 >= 0;
   y = noEvent(if valid then sqrt(y2) else 0);`
Alas, that doesn't work (at least not on the Dymola version I tried).  I didn't expect it to, but cannot say exactly why it should fail.

**Comment by beutlich** [comment]
I force-pushed to your branch with a fix that no longer raises a floating-point exception in Dymola or SimulationX.

**Pull request merged**